testsuite/features/step_definitions/common_steps.rb

```diff
@@ -371,22 +371,71 @@
 
 # setup wizard
 
-When(/^I make the credentials primary$/) do
-  raise 'xpath: i.fa-star-o not found' unless find('i.fa-star-o').click
+Then(/^HTTP proxy verification should have succeeded$/) do
+  raise 'xpath: i.text-success not found' unless find('i.text-success', wait: DEFAULT_TIMEOUT)
 end
 
-When(/^I delete the primary credentials$/) do
-  raise 'xpath: i.fa-trash-o not found' unless find('i.fa-trash-o', match: :first).click
-  step 'I click on "Delete"'
+When(/^I enter the address of the HTTP proxy as "([^"]*)"/) do |hostname|
+  step %(I enter "#{$server_http_proxy}" as "#{hostname}")
+end
+
+When(/^I ask to add new credentials$/) do
+  raise 'xpath: i.fa-plus-circle not found' unless find('i.fa-plus-circle').click
+end
+
+When(/^I enter the SCC credentials$/) do
+  scc_username = ENV['SCC_CREDENTIALS'].split('|')[0]
+  scc_password = ENV['SCC_CREDENTIALS'].split('|')[1]
+  steps %(
+    And I enter "#{scc_username}" as "edit-user"
+    And I enter "#{scc_password}" as "edit-password"
+  )
+end
+
+Then(/^the SCC credentials should be valid$/) do
+  scc_username = ENV['SCC_CREDENTIALS'].split('|')[0]
+  scc_password = ENV['SCC_CREDENTIALS'].split('|')[1]
+  within(:xpath, "//h3[contains(text(), '#{scc_username}')]/../..") do
+    raise 'Valid icon not found' unless find('i.text-success', wait: DEFAULT_TIMEOUT)
+  end
+end
+
+Then(/^the credentials for "([^"]*)" should be invalid$/) do |user|
+  within(:xpath, "//h3[contains(text(), '#{user}')]/../..") do
+    raise 'Invalid icon not found' unless find('i.text-danger', wait: DEFAULT_TIMEOUT)
+  end
+end
+
+When(/^I make the credentials for "([^"]*)" primary$/) do |user|
+  within(:xpath, "//h3[contains(text(), '#{user}')]/../..") do
+    raise 'Star icon not found' unless find('i.fa-star-o').click
```

Comment:
I would say we need to catch raise 'Star icon not found' unless find('i.fa-star-o'), without the click.
Otherwise, if the element is not found and we try to click, it will raise an exception isn't it?

Reply:
You're right. I stupidly copied the "not found" without asking myself any questions.

I don't think it makes any importance which exception it raises. I will change the message to "Unable to click on star icon".

Reply:
I would store the element found by find in a variable and click after that check... to be in the safe side, but as you wish.

Reply:
> I would store the element found by find in a variable and click after that check... to be in the safe side, but as you wish.

There's a zillion of those buggy constructs. This would deserve a big PR of its own.